src/JWT/JwtDecoder.cs

```diff
@@ -122,5 +110,28 @@ public void Validate(string[] parts, byte[] key)
 
             _jwtValidator.Validate(payloadJson, decodedCrypto, decodedSignature);
         }
+
+        /// <summary>
+        /// Get the Jwt token as string and return a parts array of <see cref="string[]"/> object.
+        /// </summary>
+        /// <param name="token">The JWT.</param>
+        /// <returns>The converted parts array of <see cref="string[]"/></returns>
+        /// <exception cref="ArgumentException">Thrown if the given token have the wrong fromat.</exception>
+        private string[] splitToParts(string token)
+        {
+            var parts = token.Split('.');
+            if (parts.Length != 3)
+            {
+                throw new ArgumentException("Token must consist from 3 delimited by dot parts");
+            }
+            return parts;
+        }
+    }
+
+    enum JwtParts

```

Comment:
I'm thinking maybe to replace it with a class representing the Jwt? Like:
```csharp
public class JwtPayload
{
    public JwtPayload(string payload) { }

    public JwtPayload(string[] parts) { }

    public string Header { get; } => _parts[0];

    public string Body { get; } => _parts[1];

    public string Signature { get; } => _parts[2];
}
```
So it'd be fine to address by index but only internally.

Reply:
Nice tip. I Resolve this comment.